src/main/java/rosenfeld/boggle/BoggleSolver.java

```diff
@@ -1,22 +1,82 @@
 package rosenfeld.boggle;
-
-import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class BoggleSolver {
-    BoggleBoard board;
-    List<String> words;
-    Game game;
-
-    public BoggleSolver (BoggleBoard boggleBoard, Game currentGame) throws IOException {
-        board = boggleBoard;
-        words = new ArrayList<>();
-        game = currentGame;
+    private BoggleBoard boggleBoard;
+    private List<String> boggleDictionary;
+    private WordTrie wordTrie;
+    private List<String> possibleWords;
+
+    public BoggleSolver(BoggleBoard boggleBoard, Game game, WordTrie wordTrie) {
+        this.boggleBoard = boggleBoard;
+        this.boggleDictionary = game.dictionary.getWordList().stream().filter(w -> w.length() > 3).collect(Collectors.toList());
+        this.wordTrie = new WordTrie();
```

Comment:
You're passing wordTrie in, and creating it in here.
Instead, only pass WordTrie in here, and make BoggleDictionary a constructor parameter to WordTrie